docs/programming/using_bobs.md

```diff
@@ -0,0 +1,149 @@
+# Working with BOBs (Blitter Objects)
+
+ACE provides a powerful system for managing and animating 2D sprites using the Amiga's hardware blitter through its BOB (Blitter OBject) system. This document explains how to use BOBs effectively in your ACE applications.
+
+## Setting Up Your Display
+
+Before using BOBs, you need to set up a proper display buffer. BOBs work best with interleaved and double-buffered simplebuffers:
+
+```c
+// Create a simple buffer viewport manager with double buffering and interleaved bitmap
+s_pVpManager = simpleBufferCreate(0,
+    TAG_SIMPLEBUFFER_BITMAP_FLAGS, BMF_CLEAR | BMF_INTERLEAVED,
+    TAG_SIMPLEBUFFER_IS_DBLBUF, 1,
+    // Add other tags as needed
+    TAG_DONE);
+```
+
+**Important**: 
+- Always use interleaved bitmaps (`BMF_INTERLEAVED`) for best performance
+- Double buffering (`TAG_SIMPLEBUFFER_IS_DBLBUF`) is essential for smooth animation
+- Make sure to export your bitmap resources as interleaved format
+
+At the end of each frame, you should process viewport managers and wait for vertical blank:
+
+```c
+// End of game loop
+viewProcessManagers(s_pView);
+copProcessBlocks();
+vPortWaitForEnd(s_pVp);
+```
+
+## What are BOBs?
+
+BOBs are 2D graphic elements that can be efficiently moved around the screen using the Amiga's hardware blitter. They offer several advantages:
+
+- Background preservation for clean undrawing
+- Transparency support via masks
+- Animation capabilities by switching frames
+- Efficient memory usage and blitter operations
+- Support for double-buffering for smooth animation
+
+## Basic BOB Workflow
+
+The typical workflow for using BOBs is:
+
+1. **Initialization** (in gamestate create):
+```c
+// Create manager with your buffers
+bobManagerCreate(pFrontBuffer, pBackBuffer, uwAvailHeight);
+
+// Initialize each BOB
+bobInit(&sBob1, uwWidth, uwHeight, 1, pFrameData, pMaskData, uwX, uwY);
+bobInit(&sBob2, uwWidth, uwHeight, 1, pFrameData, pMaskData, uwX, uwY);
+// ... initialize more BOBs as needed
+
+// Allocate memory for background saving (must be called after all BOBs are initialized)
+bobReallocateBuffers();
+```
+
+2. **Game Loop** (in gamestate loop):
+```c
+// Begin the BOB drawing cycle - undraws previous BOBs
+bobBegin();
+
+// Update BOB positions
+sBob1.sPos.uwX = newX;
+sBob1.sPos.uwY = newY;
+
+// Add BOBs to drawing queue
+bobPush(&sBob1);
+// ... perform other calculations
+bobPush(&sBob2);
+
+// Process some BOBs (save backgrounds) while doing other calculations
+bobProcessNext();
+
+// Signal no more BOBs will be added this frame
```

Comment:
```suggestion
// Signal no more BOBs will be added this frame - when using undraw buffers, it will signalize the manager that no more background saves are needed and it can start drawing BOBs on the bitmap with bobProcessNext()
```